rerun_py/tests/unit/test_expected_warnings.py

```diff
@@ -50,3 +50,14 @@ def test_expected_warnings() -> None:
         lambda: rr.log("world/image", rr.Pinhole(focal_length=3)),
         "Must provide one of principal_point, resolution, or width/height)",
     )
+    expect_warning(
+        _duplicated_recording,
+        "already exists, will ignore creation and return existing recording.",
+    )
+
+
+def _duplicated_recording() -> None:
+    """A special case of creating duplicated streams and logging information, it causes a hang, this test is for this case."""
+    application_id = "test_app_id"
+    rr.init(application_id)
+    rr.init(application_id)
```

Comment:
should inspect the recording before/after and verify that it got the same id

Reply:
should test the actually expected warning here like the other tests in this file, otherwise this is not testing anything

Reply:
the old method was all about expected warnings from log. the new one is about init. which is arguably means that it should go elsewhere, but whatever

but you didn't actually follow the pattern here either way - the pattern is using a utility method to check the error.


you're now testing `rr.init` with explicit recording ids. Which is also interesting to test, but would be nice to still test the default parameter as well. I think there's a getter on the recording for retrieving it?

Reply:
I haven't tried myself tbh, so maybe this leads nowhere, but I thought something like `get_global_data_recording().get_recording_id()` should work

Reply:
I can also take this over, it's a lot of context and I don't mean to push you from one detail to the next. Thank you for the work you're putting in already!

Reply:
But it doesn't check for the returned recording.